api/routes/news.js

```diff
@@ -2,87 +2,185 @@ const express = require('express');
 const app = express();
 app.use(express.json());
 const CustomError = require('../../error-handler/index')
-const validateNews = require('../../validators/validate')
-const newsArray = [
-    {
-        id: 1,
-        name: "Nitin"
-    },
-    {
-        id: 2,
-        name: "Slava"
-    },
-    {
-        id: 3,
-        name: "Jack"
-    },
-    {
-        id: 4,
-        name: "Kohli"
-    }
-];
+const mongoose = require('mongoose')
+const News = require('../models/news')
+const bcrypt = require('bcrypt')
+const User = require('../models/user')
+const passport = require('passport')
+
+
+
+app.get('/users/login', (req,res) => {    
+    res.render('login')
+});
+
+app.get('/users/register', (req,res) => {    
+    res.render('register')
+});
 
 app.get('/news', (req, res, next) => {
-    res.status(200).json({
-        newsArrayJSON: newsArray
-    })    
+    News.find()
+    .select('name _id')
+    .exec()
+    .then(data => {
+        if(data.length >= 10) {
+            res.status(200).json({
+                count: data.length,
+                newsArray: data                
+            })
+        } else {
+            setTimeout(function () {
```

Comment:
Inside if then lets use just throw new CustomError(/* text, params */)
It will throw an Error to the catch block

For catch blocks we can create the following function
const asyncErrorHandler = (err, next) => setTimeout(function () {
  try {
      throw new CustomError(err.message, err.status)
  } catch (err) {
    next(err)
  }
}, 0)


and use it smth like 


somePromise
  .then((data) => {
    if(!data) {
      throw new CustomError('something wrong', 500)
    }
  })
  .catch(err => asyncErrorHandler(err, next))

Reply:
Resolved